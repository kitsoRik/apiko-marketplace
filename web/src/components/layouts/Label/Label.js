import React, { useState } from 'react';

import "./Label.scss";

const Label = ({ value, as = "span", error, errorValueIfTouched, className, children, ...props }) => {

    const [touched, setTouched] = useState(false);

    if(!children) {
        return textToComponent(as, { 
            className: `label-solo-value ${className}`, error: error ? "true" : "false", ...props 
        }, value);
    }

    return ( 
        <div className={`label ${className}`} {...props} onBlur={() => setTouched(true)}>
            <div className="label-values">
                {
                    textToComponent(as, { 
                        className: `label-values-value ${className}`, error: error ? "true" : "false", ...props 
                    }, value)
                }
                {(touched || error) && <span className="label-values-error-value">{ error || errorValueIfTouched }</span>}
            </div>
            { children }
        </div>
     );
}

const textToComponent = (as, props = null, content = "") => React.createElement(as, props, content);

export default Label;