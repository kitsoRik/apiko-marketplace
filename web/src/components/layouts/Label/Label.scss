.label {
	display: grid;
	grid-template-rows: min-content auto;

	&-solo-value {
		&[error="true"] {
			color: red;
		}
	}

	&-values {
		margin-bottom: 5px;

		display: flex;

		&-value {
			text-transform: uppercase;
		}

		&-error-value {
			margin-left: 10px;
			color: red;
		}
	}
}
