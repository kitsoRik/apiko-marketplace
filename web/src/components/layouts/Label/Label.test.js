import React from "react";
import { create } from "react-test-renderer";
import Label from "./Label";

describe("Label", () => {
	it("should render without text", () => {
		const component = create(<Label />);
		const json = component.toJSON();

		expect(json).toMatchSnapshot();
	});

	it("should render with text", () => {
		const component = create(<Label value="My custom text" />);
		const json = component.toJSON();

		expect(json).toMatchSnapshot();
	});

	it("should render with text as h1", () => {
		const component = create(<Label value="My custom value" as="h1" />);
		const json = component.toJSON();

		expect(json).toMatchSnapshot();
	});

	it("should render with className", () => {
		const component = create(
			<Label value="My custom value" className="my-custom-class-name" />
		);
		const json = component.toJSON();

		expect(json).toMatchSnapshot();
	});

	it("should render with error", () => {
		const component = create(<Label error="My custom error" />);
		const json = component.toJSON();

		expect(json).toMatchSnapshot();
	});

	it("should render with error if touched", () => {
		const component = create(
			<Label errorValueIfTouched="My custom error" />
		);
		const json = component.toJSON();

		expect(json).toMatchSnapshot();
	});

	it("should render with children", () => {
		const component = create(
			<Label>
				<div>Children</div>
			</Label>
		);
		const json = component.toJSON();

		expect(json).toMatchSnapshot();
	});

	it("should render with children and value", () => {
		const component = create(
			<Label value="My custom value">
				<div>Children</div>
			</Label>
		);
		const json = component.toJSON();

		expect(json).toMatchSnapshot();
	});

	it("should render with children and error", () => {
		const component = create(
			<Label error="My custom value">
				<div>Children</div>
			</Label>
		);
		const json = component.toJSON();

		expect(json).toMatchSnapshot();
	});

	it("should render with children and error if touched", () => {
		const component = create(
			<Label errorValueIfTouched="My custom value">
				<div>Children</div>
			</Label>
		);
		const json = component.toJSON();

		expect(json).toMatchSnapshot();
	});
});
