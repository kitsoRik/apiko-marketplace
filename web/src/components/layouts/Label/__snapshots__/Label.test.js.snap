// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label should render with children 1`] = `
<div
  className="label undefined"
  onBlur={[Function]}
>
  <div
    className="label-values"
  >
    <span
      className="label-values-value undefined"
      error={null}
    >
      
    </span>
  </div>
  <div>
    Children
  </div>
</div>
`;

exports[`Label should render with children and error 1`] = `
<div
  className="label undefined"
  onBlur={[Function]}
>
  <div
    className="label-values"
  >
    <span
      className="label-values-value undefined"
      error="true"
    >
      
    </span>
    <span
      className="label-values-error-value"
    >
      My custom value
    </span>
  </div>
  <div>
    Children
  </div>
</div>
`;

exports[`Label should render with children and error if touched 1`] = `
<div
  className="label undefined"
  onBlur={[Function]}
>
  <div
    className="label-values"
  >
    <span
      className="label-values-value undefined"
      error={null}
    >
      
    </span>
  </div>
  <div>
    Children
  </div>
</div>
`;

exports[`Label should render with children and value 1`] = `
<div
  className="label undefined"
  onBlur={[Function]}
>
  <div
    className="label-values"
  >
    <span
      className="label-values-value undefined"
      error={null}
    >
      My custom value
    </span>
  </div>
  <div>
    Children
  </div>
</div>
`;

exports[`Label should render with className 1`] = `
<span
  className="label-solo-value my-custom-class-name"
  error={null}
>
  My custom value
</span>
`;

exports[`Label should render with error 1`] = `
<span
  className="label-solo-value undefined"
  error="true"
>
  
</span>
`;

exports[`Label should render with error if touched 1`] = `
<span
  className="label-solo-value undefined"
  error={null}
>
  
</span>
`;

exports[`Label should render with text 1`] = `
<span
  className="label-solo-value undefined"
  error={null}
>
  My custom text
</span>
`;

exports[`Label should render with text as h1 1`] = `
<h1
  className="label-solo-value undefined"
  error={null}
>
  My custom value
</h1>
`;

exports[`Label should render without text 1`] = `
<span
  className="label-solo-value undefined"
  error={null}
>
  
</span>
`;
